File: demos/ruby/robo_game.rb
class RoboGame
  attr_accessor :username, :email, :repo
  attr_reader :level, :game_id, :game_state

  # HTTParty stuff
  include HTTParty
  base_uri HOST 
  format :json

  def initialize(username, email, repo)
    @username = username
    @email = email
    @repo = repo
  end

  ##
  # Create a new game for the specified level. If a game already
  # exists at that level it will be overwritten.
  ##
  def start_game(level)
    @level = level
    res = self.class.post("/minefield/levels/#{@level}/games", :body => {:username => @username, :email => @email, :rep => @repo})
    @game_id = res.parsed_response["gameId"]
    @game_state = res.parsed_response["state"]
  end

  ##
  # Send a move to the game server.
  # action can be left, right, down or up 
  ##
  def move(action)
    @game_state = self.class.post("/minefield/#{@game_id}/moves", :body => {:action => action}).parsed_response
  end
end

File: demos/php/robo_game.php
<?php

class RoboGame {
    // user data
    private $username;
    private $email;
    private $repo;

    // game data
    private $level;
    private $game_id;
    private $game_state;

    function __construct($host, $username, $email, $repo) {
        $this->host = $host;
        $this->username = $username;
        $this->email = $email;
        $this->repo = $repo;
    }

    /**
     * get the current game state
     */
    public function get_game_state() {
        return $this->game_state;
    }

    /**
     * Create a new game for the specified level. If a game already exists
     * for that level it will be overwritten.
     */
    public function start_game($level) {
        $this->level = $level;

        $body = array(
            "username" => $this->username,
            "email" => $this->email,
            "repo" => $this->repo
        );

        $res = $this->post($this->host . "/minefield/levels/" . $level . "/games", $body);
        $this->game_id = $res["gameId"];
        $this->game_state = $res["state"];
    }

    /**
     * Send a move to the game server
     * $direction can be left, right, down or up
     */
    public function move($direction) {
        $body = array(
            "action" => $direction
        );

        $this->game_state = $this->post($this->host . "/minefield/" . $this->game_id . "/moves", $body);
    }


    private function post($url, $body) {
        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        $res = curl_exec($ch);
        curl_close($ch);

        return json_decode($res, true);
    }
}

?>